httpcore/_backends/asyncio.py

```diff
@@ -225,13 +225,19 @@ async def open_tcp_stream(
         port: int,
         ssl_context: Optional[SSLContext],
         timeout: TimeoutDict,
+        *,
+        local_address: Optional[str],
     ) -> SocketStream:
         host = hostname.decode("ascii")
         connect_timeout = timeout.get("connect")
         exc_map = {asyncio.TimeoutError: ConnectTimeout, OSError: ConnectError}
         with map_exceptions(exc_map):
+            local_addrport = None if local_address is None else (local_address, 0)
```

Comment:
(Nit) Consider moving this line alongside computation of `host` and `connect_timeout`?